packages/pipeline-editor/src/PipelineEditorWidget.tsx

```diff
@@ -482,47 +461,18 @@ const PipelineWrapper: React.FC<IProps> = ({
   const cleanNullProperties = React.useCallback((): void => {
     // Delete optional fields that have null value
     for (const node of pipeline?.pipelines[0].nodes) {
-      if (node.app_data.cpu === null) {
-        delete node.app_data.cpu;
+      if (node.app_data.component_parameters.cpu === null) {
+        delete node.app_data.component_parameters.cpu;
       }
-      if (node.app_data.memory === null) {
-        delete node.app_data.memory;
+      if (node.app_data.component_parameters.memory === null) {
+        delete node.app_data.component_parameters.memory;
       }
-      if (node.app_data.gpu === null) {
-        delete node.app_data.gpu;
+      if (node.app_data.component_parameters.gpu === null) {
+        delete node.app_data.component_parameters.gpu;
```

Comment:
should we enumerate all keys and delete any null entries?

Reply:
Was there a reason why only the "resource-based" parameters were deleted before this change?  Making this general will potentially delete all component parameters and I'm not sure what side effect such a change may have when considering any component (generic or custom).

Reply:
I think only resource Params are potentially null at the moment, but also wanted to double check